### PR TITLE
[21322] Support for `Fast DDS 3.0`

### PR DESCRIPTION
**Pull request by Mario-DL**
Related branches to this PR:

* [rosidl_typesupport_fastrtps - feature/support-fastdds-3.0](https://github.com/eProsima/rosidl_typesupport_fastrtps/tree/feature/support-fastdds-3.0)
* [rosidl_dynamic_typesupport_fastrtps - feature/support-fastdds-3.0](https://github.com/eProsima/rosidl_dynamic_typesupport_fastrtps/tree/feature/support-fastdds-3.0)
* ROS 2 jazzy over Ubuntu 24.04

This PR adds support for `Fast DDS 3.0`  to `rmw_fastrtps_shared_cpp`, `rmw_fastrtps_cpp` and `rmw_fastrtps_dynamic_cpp`.

~~[demo_nodes_cpp](https://github.com/ros2/demos/) is (semi) functional (`jazzy` branch, `rmw_fastrtps_cpp` rmw) but we need to review some errors occasionally arising in the `[TYPELOOKUP_SERVICE_REPLY_LISTENER] Error registering remote type: TypeIdentifier unknown to TypeObjectRegistry` 
when `talker` and `listener` are launched (it triggers always in the secondly launched executable) making the application to crash.~~
With latest changes and https://github.com/eProsima/Fast-DDS/pull/5181, demos are functional.

~~Preliminar analysis shows that the problem is who registers the types. If the rmw is fast enough to register the types, then the demo works fine. However if the types are registered from the typelookup service first, some of them are missing or incomplete and leads to crash.
In addition, an arguable behavior was also noticed in the `TypeObjectRegistry::register_type_object`. Imagine that the TLS registers an (incomplete, wrong or corrupt) type object for a given type name. Then, the rmw would like to register the typeobject again but instead of making a logic to accept, discard or update the existing type object in the registry, the typeidentifiers and are instead removed from the collections. This behavior could be reviewed although it most probably be a thing to be taken into consideration apart from the real fix, as it would mitigate the "consequence" instead of the root cause of the problem, probably lying in the TypeLookup service instead.~~  

Addressed in https://github.com/eProsima/Fast-DDS/pull/5181

~~Finally, not all TODOs were address, but theoretically the remaining ones not affecting the core behavior.~~

### CODE REVIEW

**Review by richiware — CHANGES_REQUESTED** [review]
Partial review. Pending `TypeSupport_impl.cpp` to be reviewed.

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp [hunk 1]
```diff
@@ -500,132 +482,166 @@ MemberIdentifierName GetTypeIdentifier(const MembersType * members, uint32_t ind
...
-  bool ret = false;
   if (type_support_intro->typesupport_identifier ==
     rosidl_typesupport_introspection_c__identifier)
   {
-    ret = add_type_object<rosidl_typesupport_introspection_c__MessageMembers>(
-      type_support_intro->data, type_name);
+    type_identifiers_ =
+      register_type_identifiers<rosidl_typesupport_introspection_c__MessageMembers>(
+      type_support_intro->data, get_name());
+    // TODO(richiware) is the name set here in TypeSupport in ROS2? check.
```
Comment (richiware) [inline rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:639]: This `TODO` can be removed. I checked it.

**Review by richiware — CHANGES_REQUESTED** [review]

**Review by richiware — CHANGES_REQUESTED** [review]
`cpplint` fails

File: rmw_fastrtps_cpp/package.xml [hunk 2]
```diff
@@ -21,7 +21,7 @@
   <buildtool_export_depend>ament_cmake</buildtool_export_depend>
 
   <build_depend>fastcdr</build_depend>
-  <build_depend>fastrtps</build_depend>
+  <build_depend>fastdds</build_depend>
   <build_depend>fastrtps_cmake_module</build_depend>
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply rmw_fastrtps_cpp/package.xml:25]: As we internally agree we will rename it when we unify the rmws in a followup

File: rmw_fastrtps_cpp/rmw_fastrtps_cpp-extras.cmake [hunk 3]
```diff
@@ -16,9 +16,9 @@
 
 find_package(fastrtps_cmake_module REQUIRED)
 find_package(fastcdr 2 REQUIRED CONFIG)
-find_package(fastrtps 2.13 REQUIRED CONFIG)
-find_package(FastRTPS 2.13 REQUIRED MODULE)
+find_package(fastdds 3 REQUIRED CONFIG)
+find_package(FastDDS 3 REQUIRED MODULE)
 
-list(APPEND rmw_fastrtps_cpp_INCLUDE_DIRS ${FastRTPS_INCLUDE_DIR})
+list(APPEND rmw_fastrtps_cpp_INCLUDE_DIRS ${FAST_INCLUDE_DIR})
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply rmw_fastrtps_cpp/rmw_fastrtps_cpp-extras.cmake:22]: As we internally agree we will rename it when we unify the rmws in a followup

**Review by richiware — APPROVED** [review]

**Pull request merged**